tests/destructure.test.ts

```diff
@@ -0,0 +1,228 @@
+    // a : int = 1
+    // b : bool = True
+    // c : [int] = []
+    // d : int = 2
+    // a, *c, b, d = 2, 4
+    // `, NONE);
+
+    assertTC("destructure-assignment-with-range", `
+    ${rangeDef}
+    a : int = 0
+    b : int = 0
+    a, _, b = range(1, 4)
+    `, NONE);
+
+    assertTCFail("destructure-assignment-with-range-type-failure", `
+    ${rangeDef}
+    a : int = 0
+    b : bool = False
+    a, _, b = range(1, 4)
+    `);
+
+    assertTC("destructure-assignment-in-func", `
+    ${rangeDef}
+    def func(a: int, b: int)->iterator:
+        i : int = 0
+        j : int = 0
+        i, _, _, j = range(a, b)
+        return range(i, j)
+    a : int = 0
+    b : int = 0
+    a, _, b = func(1, 5)
+    `, NONE);
+
+    assertPrint("destructure-assignment-sep", `
+    a : int = 1
+    b : bool = True
+    a, b = 2, False
+    print(a)
+    print(b)
+    `, ['2', 'False']);
+
+    assertPrint("destructure-assignment-with-ignore-sep", `
+    a : int = 1
+    b : bool = True
+    a, _, b = 2, 3, False
+    print(a)
+    print(b)
+    `, ['2', 'False']);
+
+    assertPrint("destructure-assignment-with-range-sep", `
+    ${rangeDef}
+    a : int = 0
+    b : int = 0
+    a, b = range(1, 3)
+    print(a)
+    print(b)
+    `, ['1', '2']);
+
+    assertPrint("destructure-assignment-in-func-sep", `
+    ${rangeDef}
+    def func(a: int, b: int)->iterator:
+        i : int = 0
+        j : int = 0
+        i, _, j = range(a, b)
+        return range(i, j)
+    a : int = 0
+    b : int = 0
+    a, b = func(1, 4)
+    print(a)
+    print(b)
+    `, ['1', '2']);
+
+    assertPrint("destructure-assignment-in-class-sep", `
+    ${rangeDef}
+    c: cl = None
+  
+    class cl(Object):
+        f1: int = 0
+        f2: bool = False
+    
+    c = cl()
+    c.f1, c.f2 = 1, True
+    print(c.f1)
+    print(c.f2)
+    `, ['1', 'True']);
+
+    assertPrint("destructure-assignment-in-class-range-sep", `
+    ${rangeDef}
+    c: cl = None
+  
+    class cl(Object):
+        f1: int = 0
+        f2: int = 0
+    
+    c = cl()
+    c.f1, c.f2 = range(1, 3)
+    print(c.f1)
+    print(c.f2)
+    `, ['1', '2']);
+
+    assertTCFail("destructure-assignment-in-class-type-failure", `
+    ${rangeDef}
+    c: cl = None
+  
+    class cl(Object):
+        f1: bool = 0
+        f2: bool = 0
+    
+    c = cl()
+    c.f1, c.f2 = range(1, 3)
+    `);
+
+    assertTCFail("destructure-assignment-in-none-obj-failure", `
+    ${rangeDef}
+    c: cl = None
+  
+    class cl(Object):
+        f1: bool = 0
+        f2: bool = 0
+    
+    c.f1, c.f2 = range(1, 3)
+    `);
+
+    assertTCFail("destructure-assignment-in-not-class-failure", `
+    ${rangeDef}
+    c : int = 0
+    c.f1, c.f2 = range(1, 3)
+    `);
+
+    assertPrint("destructure-assignment-in-mix-sep", `
+    ${rangeDef}
+    c: cl = None
+    a: int = 0
+  
+    class cl(Object):
+        f1: int = 0
+    
+    c = cl()
+    c.f1, _, a = range(1, 4)
+    print(c.f1)
+    print(a)
+    `, ['1', '3']);
+    
+});
```

Comment:
Great tests, should we have some that check length mismatch in assign? something like:  `a,b = 1,2,3` or `a,b,c = 1`
Also going above literals, like `a,b,c = 1 , f(12), g(4)+5`

Reply:
Thank you for comments! We do have one test for `a, b = 2`, and we can add more tests like `a, b = 2, 3, 4` and the case you mentioned.

Reply:
> Great tests, should we have some that check length mismatch in assign? something like: `a,b = 1,2,3` or `a,b,c = 1` Also going above literals, like `a,b,c = 1 , f(12), g(4)+5`

done!